service-base/src/main/java/org/eclipse/hono/service/management/tenant/Tenant.java

```diff
@@ -166,12 +166,16 @@ public final Tenant setDefaults(final Map<String, Object> defaults) {
      *
      * @param adapters The configuration properties.
      * @return This instance, to allow chained invocations.
+     * @throws IllegalArgumentException if the adapters list is empty or {@code null}
      */
     public final Tenant setAdapters(final List<Adapter> adapters) {
-        this.adapters.clear();
-        if (adapters != null) {
-            this.adapters.addAll(adapters);
+
+        if (adapters == null || adapters.isEmpty()) {
```

Comment:
I do not quite see why we need to prevent `null`here. IMHO setting to `null` should simply clear the list (as before).

Reply:
@sophokles73 Done.